diff --git a/net/openvpn-easy-rsa/Makefile b/net/openvpn-easy-rsa/Makefile
index b35937ca2..128b01792 100644
--- a/net/openvpn-easy-rsa/Makefile
+++ b/net/openvpn-easy-rsa/Makefile
@@ -9,11 +9,11 @@ include $(TOPDIR)/rules.mk

 PKG_NAME:=openvpn-easy-rsa

-PKG_VERSION:=3.0.8
-PKG_RELEASE:=4
+PKG_VERSION:=3.1.3
+PKG_RELEASE:=1
 PKG_SOURCE_URL:=https://codeload.github.com/OpenVPN/easy-rsa/tar.gz/v$(PKG_VERSION)?
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
-PKG_HASH:=fd6b67d867c3b8afd53efa2ca015477f6658a02323e1799432083472ac0dd200
+PKG_HASH:=f2967aa598cb603dd20791002e767d0ce58e300b04c9cff1b6d6b14fedae6a80

 # For git snapshots
 #PKG_SOURCE_PROTO:=git
