From aff8367c7afb8decef9e52a0ebf4dea72a16b39b Mon Sep 17 00:00:00 2001
From: hingbong lo <hingbonglo@gmail.com>
Date: Fri, 10 Jan 2025 00:50:53 +0000
Subject: [PATCH] rust: update to 1.84.0

Signed-off-by: hingbong lo <hingbonglo@gmail.com>
---
 lang/rust/Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lang/rust/Makefile b/lang/rust/Makefile
index b4b6285e7de2e3..ab5436741230fe 100644
--- a/lang/rust/Makefile
+++ b/lang/rust/Makefile
@@ -5,12 +5,12 @@
 include $(TOPDIR)/rules.mk

 PKG_NAME:=rust
-PKG_VERSION:=1.81.0
+PKG_VERSION:=1.85.0
 PKG_RELEASE:=1

 PKG_SOURCE:=rustc-$(PKG_VERSION)-src.tar.gz
 PKG_SOURCE_URL:=https://static.rust-lang.org/dist/
-PKG_HASH:=872448febdff32e50c3c90a7e15f9bb2db131d13c588fe9071b0ed88837ccfa7
+PKG_HASH:=2f4f3142ffb7c8402139cfa0796e24baaac8b9fd3f96b2deec3b94b4045c6a8a
 HOST_BUILD_DIR:=$(BUILD_DIR)/host/rustc-$(PKG_VERSION)-src

 PKG_MAINTAINER:=Luca Barbato <lu_zero@luminem.org>
