From 4544ce5cb2079c110808d10d5ae45d002f4cfc8f Mon Sep 17 00:00:00 2001
From: lunatickochiya <125438787@qq.com>
Date: Sun, 6 Oct 2024 09:31:28 +0800
Subject: [PATCH] mdnsd:std err

---
 net/mdnsresponder/files/mdnsd.init | 1 +
 1 file changed, 1 insertion(+)

diff --git a/net/mdnsresponder/files/mdnsd.init b/net/mdnsresponder/files/mdnsd.init
index 817604b..8f73937 100755
--- a/net/mdnsresponder/files/mdnsd.init
+++ b/net/mdnsresponder/files/mdnsd.init
@@ -9,6 +9,7 @@ start_service() {
     procd_open_instance
     procd_set_param command /usr/sbin/mdnsd -debug
     procd_set_param respawn
+    procd_set_param stderr 1
     procd_close_instance
 }
 
-- 
2.34.1

