diff --git a/net/aria2/Makefile b/net/aria2/Makefile
index cfac620..9a48788 100644
--- a/net/aria2/Makefile
+++ b/net/aria2/Makefile
@@ -108,5 +108,9 @@ define Package/aria2/install
 	$(INSTALL_CONF) ./files/aria2.conf $(1)/etc/config/aria2
 endef

+define Package/aria2-openssl/install
+	$(INSTALL_DIR) $(1)/root
+endef
+
 $(eval $(call BuildPackage,aria2))
 $(eval $(call BuildPackage,aria2-openssl))
