diff --git a/package/utils/util-linux/Makefile b/package/utils/util-linux/Makefile
index 90724eb3f6..7497f92f40 100644
--- a/package/utils/util-linux/Makefile
+++ b/package/utils/util-linux/Makefile
@@ -452,6 +452,7 @@ define Package/script-utils
 $(call Package/util-linux/Default)
   TITLE:=make and replay typescript of terminal session
   SUBMENU=Terminal
+  ALTERNATIVES:=200:/usr/bin/script:/usr/bin/util-linux-script
 endef
 
 define Package/script-utils/description
@@ -846,7 +847,7 @@ endef
 
 define Package/script-utils/install
 	$(INSTALL_DIR) $(1)/usr/bin
-	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/bin/script $(1)/usr/bin/
+	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/bin/script $(1)/usr/bin/util-linux-script
 	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/bin/scriptreplay $(1)/usr/bin/
 endef
 
