From 9b8ba992ae647bec86689eaa603d9929b2b6cf4c Mon Sep 17 00:00:00 2001
From: lunatickochiya <125438787@qq.com>
Date: Fri, 20 Dec 2024 22:02:06 +0800
Subject: [PATCH 6/9] remove default uboot-rk35xx

---
 package/boot/uboot-rk35xx/Makefile | 1 -
 1 file changed, 1 deletion(-)

diff --git a/package/boot/uboot-rk35xx/Makefile b/package/boot/uboot-rk35xx/Makefile
index 9de0d65618..9c1067cbd7 100644
--- a/package/boot/uboot-rk35xx/Makefile
+++ b/package/boot/uboot-rk35xx/Makefile
@@ -29,7 +29,6 @@ define U-Boot/Default
   UENV:=default
   HIDDEN:=1
   BUILD_SUBTARGET:=rk35xx
-  DEFAULT:=y
   PKG_BUILD_DEPENDS:=arm-trusted-firmware-rockchip-vendor
 endef
 
-- 
2.34.1

