From 6269857ae90df89086281e1f11ef6d7f63c290e1 Mon Sep 17 00:00:00 2001
From: lunatickochiya <125438787@qq.com>
Date: Sat, 21 Dec 2024 12:24:50 +0800
Subject: [PATCH 8/9] vendor:add rk3399 rk3328 support

---
 .../Makefile                                  | 23 +++++++++++++------
 .../atf-version.mk                            |  8 +++++++
 2 files changed, 24 insertions(+), 7 deletions(-)

diff --git a/package/boot/arm-trusted-firmware-rockchip-vendor/Makefile b/package/boot/arm-trusted-firmware-rockchip-vendor/Makefile
index 17585ba44e..9fbfd8c07c 100644
--- a/package/boot/arm-trusted-firmware-rockchip-vendor/Makefile
+++ b/package/boot/arm-trusted-firmware-rockchip-vendor/Makefile
@@ -30,10 +30,18 @@ define Trusted-Firmware-A/rk3308
   DDR1:=rk33/$(RK3308_DDR_UART2)
 endef
 
-define Trusted-Firmware-A/rk3528
-  BUILD_SUBTARGET:=rk35xx
-  ATF:=rk35/$(RK3528_ATF)
-  DDR:=rk35/$(RK3528_DDR)
+define Trusted-Firmware-A/rk3328
+  BUILD_SUBTARGET:=armv8
+  ATF:=rk33/$(RK3328_ATF)
+  DDR:=rk33/$(RK3328_DDR)
+  LOADER:=rk33/$(RK3328_LOADER)
+endef
+
+define Trusted-Firmware-A/rk3399
+  BUILD_SUBTARGET:=armv8
+  ATF:=rk33/$(RK3399_ATF)
+  DDR:=rk33/$(RK3399_DDR)
+  LOADER:=rk33/$(RK3399_LOADER)
 endef
 
 define Trusted-Firmware-A/rk3568
@@ -46,12 +54,13 @@ define Trusted-Firmware-A/rk3588
   BUILD_SUBTARGET:=rk35xx
   ATF:=rk35/$(RK3588_ATF)
   DDR:=rk35/$(RK3588_DDR)
-  DDR1:=rk35/$(RK3588_DDR_LOWFREQ)
-endef
+  DDR1:=rk35/$(RK3588_DDR_LOWFREQ)endef
 
 TFA_TARGETS:= \
 	rk3308 \
 	rk3528 \
+	rk3328 \
+	rk3399 \
 	rk3568 \
 	rk3588
 
@@ -87,4 +96,4 @@ endif
 endif
 endef
 
-$(eval $(call BuildPackage/Trusted-Firmware-A))
\ No newline at end of file
+$(eval $(call BuildPackage/Trusted-Firmware-A))
diff --git a/package/boot/arm-trusted-firmware-rockchip-vendor/atf-version.mk b/package/boot/arm-trusted-firmware-rockchip-vendor/atf-version.mk
index 18f3543c63..5d3cd16327 100644
--- a/package/boot/arm-trusted-firmware-rockchip-vendor/atf-version.mk
+++ b/package/boot/arm-trusted-firmware-rockchip-vendor/atf-version.mk
@@ -11,3 +11,11 @@ RK3568_DDR:=rk3568_ddr_1560MHz_v1.16.bin
 RK3588_ATF:=rk3588_bl31_v1.37.elf
 RK3588_DDR:=rk3588_ddr_lp4_2112MHz_lp5_2736MHz_v1.10.bin
 RK3588_DDR_LOWFREQ:=rk3588_ddr_lp4_1866MHz_lp5_2736MHz_v1.10.bin
+
+RK3328_ATF:=rk322xh_bl31_v1.49.elf
+RK3328_DDR:=rk3328_ddr_333MHz_v1.19.bin
+RK3328_LOADER:=rk322xh_miniloader_v2.50.bin
+
+RK3399_ATF:=rk3399_bl31_v1.36.elf
+RK3399_DDR:=rk3399_ddr_800MHz_v1.30.bin
+RK3399_LOADER:=rk3399_miniloader_v1.26.bin
-- 
2.34.1

